.github/workflows/build-release.yml

```diff
@@ -0,0 +1,29 @@
+name: Build and Release
+
+on:
+  release:
+    types: [ released ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Validate composer.json and composer.lock
+        run: composer validate --strict
+
+      - name: Install dependencies
+        run: |
+          composer run-script packages-install -- --no-dev
+          npm ci
+          npm run build
+
+      - name: Zip Folder
```

Comment:
Could we use 10up's script to do this? They have a `generate-zip` option which sounds like it could replace this entire thing. Maybe we can minify the whole thing to one single workflow that does different things based on whether the release tag is a beta, pre-release, or stuff like that...

Reply:
I considered this but ultimately decided not to. I'm hesitant to do this mostly because we're not normally deploying to .org, but we do create new releases for some plugins for partner sites - and hence I didn't want the two workflows tied together. Also, 51 devs and (some?) TAMs typically only create releases, ie no pre-releases, beta, etc, and changing that creates another opportunity for "accidental" deploys if it becomes too complex, and so I wanted the deploy to .org be intentional. 

I'm fine with changing this approach and consolidating down to a single/or fewer workflows if you think that's the better approach.